### ISSUE TITLE
Changing scales on a pattern should transform the scale

### ISSUE THREAD

**Issue by TerryCavanagh** [TerryCavanagh/boscaceoil#48]
This would be consistent with the behaviour when changing keys, and also make some cool creative stuff possible!

**Comment by RFDaemoniac** [comment]
I've been playing around with bosca ceoil as a learning tool and changing scales, and would like to see notes not deleted from lines that were visible on the previous scale but not visible on the new scale. So that you could limit a piece to a single scale to see how it changes it (essentially just a filter on what notes show and play rather than deleting them)

But I understand that this might not be super intuitive/as good of an answer as the above.

**Comment by TerryCavanagh** [comment]
Yeah, I think that would be the other logical way to approach this. Thanks for the comment! Will keep it in mind when I look at implementing this in future
